### ISSUE TITLE
exporting the pix2pix model to onnx model, but the inference result by onnx is incorrect

### ISSUE THREAD

**Issue by shingming** [junyanz/pytorch-CycleGAN-and-pix2pix#1276]
Recently, I need to export the pix2pix model to onnx in order to deploy that to other applications.

I ref. the transformation code from this post: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/1113#issuecomment-728681741

Although I successfully convert the pix2pix model to onnx, I get the incorrect result by the onnx model compare to the pth model output in the same input.

I survey the [paper](https://arxiv.org/pdf/1611.07004.pdf), and I got some helpful info.:

1. **Instead, for our final models, we provide noise only in the form of dropout, applied on several layers of our generator at both training and test time. Despite the dropout noise, we observe only minor stochasticity in the output of our nets.**

2. **At inference time, we run the generator net in exactly the same manner as during the training phase. This differs from the usual protocol in that we apply dropout at test time, and we apply batch normalization [26] using the statistics of the test batch, rather than aggregated statistics of the training batch.**

That means we can't disable the dropout layer when transferring the model from pth to onnx. So, I set the training argument `training=TrainingMode.TRAINING` in export API and `model.training=True`. Unfortunately, I got the same result after that. I have no idea why... 

The following images are input data and result from the onnx model and pth model.

|  input data x  | pth model output y (expected result) | onnx model output y (incorrect result) |
| ------------- | ------------- |  ------------- |
| ![input data x](https://user-images.githubusercontent.com/65477270/115984713-29a48200-a5db-11eb-9bcd-3d8dc02b4428.PNG)  | ![pth y](https://user-images.githubusercontent.com/65477270/115984786-78521c00-a5db-11eb-9158-51fe69649b6b.png)  | ![onnx y](https://user-images.githubusercontent.com/65477270/115984794-7daf6680-a5db-11eb-90d6-dd9e1b642564.PNG)  |

**Comment by mrljwlm** [comment]
the same problem with CycleGAN using 'unet_256', any one can give me some help?

**Comment by synthetica3d** [comment]
If you train with --norm instance the problem should go away. I noticed that if you run the test program after calling eval() on netG, the result will also be wrong. This is very strange because the dropout in batch norm is supposed to behave differently in train and inference, but pix2pix seems to want the exact same behavior in both training and inference. Since onnx is by definition for inference only, it is possible that batch norm, behaving as expected for inference, breaks everything. Anyway, using batch instance removes the problem.

**Comment by tongchangD** [comment]
are You try use [faceenhancement](https://github.com/microsoft/Bringing-Old-Photos-Back-to-Life/tree/master/Face_Enhancement) model to onnx

**Comment by Bea07** [comment]
I got same problems and calling eval() is the reason.

**Comment by Bea07** [comment]
> If you train with --norm instance the problem should go away. I noticed that if you run the test program after calling eval() on netG, the result will also be wrong. This is very strange because the dropout in batch norm is supposed to behave differently in train and inference, but pix2pix seems to want the exact same behavior in both training and inference. Since onnx is by definition for inference only, it is possible that batch norm, behaving as expected for inference, breaks everything. Anyway, using batch instance removes the problem.

@synthetica3d onnx have problems with instance normalization.
